axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/AggregateBasedAxonServerEventStorageEngine.java

```diff
@@ -204,13 +202,19 @@ private MessageStream<EventMessage<?>> eventsForCriterion(SourcingCondition cond
         return MessageStream.fromStream(
                 aggregateStream.asStream(),
                 this::convertToMessage,
-                event -> Context.with(LegacyResources.AGGREGATE_IDENTIFIER_KEY, event.getAggregateIdentifier())
-                                .withResource(LegacyResources.AGGREGATE_TYPE_KEY, event.getAggregateType())
-                                .withResource(LegacyResources.AGGREGATE_SEQUENCE_NUMBER_KEY,
-                                              event.getAggregateSequenceNumber())
-                                .withResource(ConsistencyMarker.RESOURCE_KEY,
-                                              new AggregateBasedConsistencyMarker(event.getAggregateIdentifier(),
-                                                                                  event.getAggregateSequenceNumber())));
+                AggregateBasedAxonServerEventStorageEngine::eventMessageContext
+        );
+    }
+
+    private static Context eventMessageContext(Event event) {
+        Context legacyContext =
+                Context.with(LegacyResources.AGGREGATE_IDENTIFIER_KEY, event.getAggregateIdentifier())
+                       .withResource(LegacyResources.AGGREGATE_TYPE_KEY, event.getAggregateType())
+                       .withResource(LegacyResources.AGGREGATE_SEQUENCE_NUMBER_KEY, event.getAggregateSequenceNumber());
```

Comment:
Seeing this part, immediately I thought; great, now we have a 5-layer ProcessingContext of each only 1 resource.  That's going to be less than fun in the debugger, and the toString will also be less than optimal to read.

I do concede that this `withResource` is handy for branching (mutating the ProcessingContext without it getting affected in other branches), but maybe we should make a distinction and have two methods. One, `withResource`, would just put the resource in it and return itself. And then we have a `branch()` method, which will create a decorating one around is. wdyt?

Of course, this is not required for this PR, just wanted to get it out there

Reply:
Ahh my bad, yes, you seem to be correct. In this case, it would merge the resources.
Resolving the comment for now, although my original point I will still inject into the meeting ;)